backend/src/main/java/org/example/follow/controller/FollowApiController.java

```diff
@@ -0,0 +1,48 @@
+package org.example.follow.controller;
+
+import org.example.exception.common.ApiErrorCategory;
+import org.example.exception.user.ApiUserErrorSubCategory;
+import org.example.exception.user.ApiUserException;
+import org.example.follow.domain.dto.FollowRequestDto;
+import org.example.follow.service.FollowService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/me/follow")
+public class FollowApiController {
+
+	private final FollowService followService;
+
+	/**
+	 * 팔로우, 언팔로우
+	 */
+	@PutMapping("")
+	public ResponseEntity<?> handleFollow(
+		@RequestBody FollowRequestDto requestDto
+	) {
+		String fromUser = SecurityContextHolder.getContext().getAuthentication().getName();
+		String toUser = requestDto.getNickname();
+
+		switch (requestDto.getFollowStatus()) {
```

Comment:
Unfollow, follow 2개가 필요한거는 이해하는데 굳이 switch문을 사용해서 default에 예외를 터트리는 이유가 궁금합니다

Reply:
@limbaba1120
맞는 지적입니다.
status가 2개 이상일 경우가 없어서 switch를 안써도 됩니다.
그러면 아래와 같이 작성 됩니다.

```suggestion
		if (requestDto.followStatus().equals(FollowStatus.FOLLOW)) {
			followService.follow(toUser, fromUser);
		} else if (requestDto.followStatus().equals(FollowStatus.UNFOLLOW)) {
			followService.unFollow(fromUser, toUser);
		} else {
			throw ApiUserException
				.builder()
				.category(ApiErrorCategory.RESOURCE_BAD_REQUEST)
				.subCategory(ApiUserErrorSubCategory.USER_FOLLOW_INVALID_REQUEST)
				.setErrorData(() -> String.format("follow status = %s", requestDto.followStatus()))
				.build();
		}
        return ResponseEntity.ok().build();
```
기존 Switch (case, break 방식) 문이였으면 사용 안했을 것 같은데, 이 경우
위와 같은 swicth expression으로 짜면 코드가 한눈에 보여서 더 가독성이 좋다고 생각합니다.
이어서 의견 주시면 감사하겠습니다.